flink-runtime/src/test/java/org/apache/flink/runtime/testutils/CommonTestUtils.java

```diff
@@ -213,6 +216,19 @@ public static void waitForAllTaskRunning(
                 timeout);
     }
 
+    public static void waitForNoTaskRunning(
+            SupplierWithException<JobDetailsInfo, Exception> jobDetailsSupplier, Deadline timeout)
+            throws Exception {
+        waitUntilCondition(
+                () -> {
+                    final Map<ExecutionState, Integer> state =
+                            jobDetailsSupplier.get().getJobVerticesPerState();
+                    return state.get(ExecutionState.RUNNING).equals(0);
```

Comment:
Couldn't that potentially give an NPE? Maybe

```suggestion
                    return state.get(ExecutionState.RUNNING) == null || state.get(ExecutionState.RUNNING).equals(0);
```